packages/utilities/src/IFabricConfig.ts

```diff
@@ -0,0 +1,26 @@
+import { IStyleSheetConfig } from '@uifabric/merge-styles';
+
+/**
+ * The interface of window.FabricConfig, which can be burned on the page before script loading to preemptively
+ * define default configurations.
+ * {@docCategory IFabricConfig}
+ */
+export interface IFabricConfig {
+  /**
+   * An override for where the fonts should be downloaded from.
+   */
+  fontBaseUrl?: string;
+
+  /**
+   * The mergeStyles stylesheet config.
+   */
+  mergeStyles?: IStyleSheetConfig;
+
+  /**
+   * A list of features to enable.
+   * A feature is enabled if the name associated with it is in the list and is set to "true".
+   * `name`s MUST be globally unique.
+   * See further documentation at https://github.com/microsoft/fluentui/wiki/enabledFeatures-Switches
+   */
+  enabledFeatures?: { [name: string]: boolean };
```

Comment:
A few thoughts

1. I like that utilities is decoupled from knowing the flags available.
2. I don't like that a user has no clue about which flags are available.
3. Please, no guids in the code. Not even in examples.

Overall I'm not sure of a cleaner solution, other than:

OUFR package exports IFabricConfig which redefines `enabledFeatures` with type safety including flags that apply to the suite.

Reply:
Type safety would mean that code enabling features will break once we graduate the feature, though, which is explicitly something we're trying to avoid.

Reply:
Once we graduate a feature, we have many options. In a major release we default the feature to on and mark the flag as deprecated. Things still build, but there's a whole major release to remove the flag. In the following major, we remove the flag. You clean up your code.

I don't think "no typings" is the way to make the flags useful. Nobody will know what flags can be used at all, without parsing all the code for secrets.

Reply:
Another idea would be along the lines of a provider:

@fluentui/react-feature-flags package, super lightweight, no type safety.

```tsx
export const FeatureFlagsContext = React.context<any>({});

export type FeaturesProviderType = <TProps>(props: TProps) => JSX.Element;
export const BaseFeaturesProvider = function<TProps = {}>(props: TProps) => {
  const { children, ...rest } = props;
 
 return (
  <FeatureFlagsContext.Provider value={rest}>{children}<FeatureFlagsContext.Provider>
  );
};

export const useBaseFeatureFlags = function<TProps> (): TProps => useContext(FeatureFlagsContext);
```

Now a package, like `@fluentui/react` can re-export this and provide the flag definitions:

```tsx
import { BaseFeaturesProvider } from '@fluentui/react-feature-flags';

// Type safety and intellisense over what features I can enable!
export interface FeaturesProps {
  flag1?: boolean;
  flag2?: boolean;
}

// Re export with type safety. Easy to combine types if managing multiple sub package features.
export const FeaturesProvider: FeaturesProviderType<FeaturesProps> = BaseFeaturesProvider;
export const FeaturesConsumer: FeaturesConsumerType<FeaturesProps> = FeaturesContext.Consumer;
export const useFeatures: FeaturesHook<FeaturesProps> = useBaseFeatureFlags;
```

Component author experience is easy:

```tsx
import { useFeatureFlags } from '../...';

const Component = (...) => {
  const { flag1 } = useFeatureFlags();

  ...do something with flag1
};

App developer experience is easy:

<FeaturesProvider flag1>
  <App .. />
</FeaturesProvider>
```
No global references or side effects.

This layering can abstract any behavior we want. Flags could be represented as an object as well rather than a boolean, so that additional metadata like generated warnings, owner names, dates, etc and such could be provided.

Would this work? Ultimately I think we want to shoot for less reliance on globals and window objects (especially with the word Fabric, when we are working on renaming) and undocumented, magic flags that are not type safe are all things that create confusion and debt.